runtime/oti/VMHelpers.hpp

```diff
@@ -2048,13 +2049,73 @@ class VM_VMHelpers
 #endif /* JAVA_SPEC_VERSION > 11 */
 	}
 
+#if JAVA_SPEC_VERSION >= 19
+/*
+ *	If low tagging failed (or not even attempted) due to either
+ *
+ *   a carrier thread winning to mount, we don't need to do anything, since it will be compensated by pre/post mount actions
+ *   another GC thread winning to scan, again don't do anything, and let the winning thread do the work, instead
+ */
+	static VMINLINE bool
+	tryWinningConcurrentGCScan(J9VMContinuation *continuation)
+	{
+		volatile uintptr_t *localAddr = &continuation->state;
+		uintptr_t oldValue = *localAddr;
+
+		if (0 == oldValue) {
+			/* atomically low tag ('set', not 'or') continuation->state */
+			if (VM_AtomicSupport::lockCompareExchange(localAddr, oldValue, J9_GC_CONTINUATION_CONCURRENTSCANNING) == oldValue) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	static VMINLINE J9VMThread *
+	getCarrierThreadFromContinuationState(uintptr_t continuationState)
+	{
+		return (J9VMThread *) (continuationState & (~(uintptr_t)J9_GC_CONTINUATION_CONCURRENTSCANNING));
+	}
+
+	static VMINLINE bool
+	isConcurrentlyScanned(uintptr_t continuationState)
+	{
+		return J9_ARE_ANY_BITS_SET(continuationState, J9_GC_CONTINUATION_CONCURRENTSCANNING);
+	}
+
+	static VMINLINE void
+	exitConcurrentGCScan(J9VMContinuation *continuation)
+	{
+		volatile uintptr_t *localAddr = &continuation->state;
+
+		/* clear CONCURRENTSCANNING flag */
+		uintptr_t oldContinuationState = VM_AtomicSupport::bitAnd(localAddr, ~(uintptr_t)J9_GC_CONTINUATION_CONCURRENTSCANNING);
+		if (J9_ARE_ANY_BITS_SET(oldContinuationState, ~(uintptr_t)J9_GC_CONTINUATION_CONCURRENTSCANNING)) {
+			J9VMThread *carrierThread = getCarrierThreadFromContinuationState(oldContinuationState);
+			omrthread_monitor_enter(carrierThread->publicFlagsMutex);
+			/*  notify the waiting carrierThread that we just finished scanning, and it can proceed with mounting. */
+			omrthread_monitor_notify(carrierThread->publicFlagsMutex);
+			omrthread_monitor_exit(carrierThread->publicFlagsMutex);
+		}
+	}
+#endif /* JAVA_SPEC_VERSION >= 19 */
+
 	static VMINLINE UDATA
-	walkContinuationStackFramesWrapper(J9VMThread *vmThread, j9object_t continuationObject, J9StackWalkState *walkState)
+	walkContinuationStackFramesWrapper(J9VMThread *vmThread, j9object_t continuationObject, J9StackWalkState *walkState, bool syncWithConcurrentGCScan)
```

Comment:
as mentioned earlier, this should be renamed to syncWithContinuationMounting (and any flags in between the 2 spots)